packages/AdaptiveCards/Actions/GetAdaptiveCardTemplate.cs

```diff
@@ -0,0 +1,209 @@
+        public const string Kind = "Microsoft.Bot.Components.GetAdaptiveCardTemplate";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GetAdaptiveCardTemplate"/> class.
+        /// </summary>
+        /// <param name="callerPath">Optional, source file full path.</param>
+        /// <param name="callerLine">Optional, line number in source file.</param>
+        public GetAdaptiveCardTemplate([CallerFilePath] string callerPath = "", [CallerLineNumber] int callerLine = 0)
+            : base()
+        {
+            this.RegisterSourceLocation(callerPath, callerLine);
+        }
+
+        /// <summary>
+        /// Gets or sets an optional expression which if is true will disable this action.
+        /// </summary>
+        /// <example>
+        /// "user.age > 18".
+        /// </example>
+        /// <value>
+        /// A boolean expression. 
+        /// </value>
+        [JsonProperty("disabled")]
+        public BoolExpression Disabled { get; set; }
+
+        [JsonProperty("location")]
+        public StringExpression Location { get; set; }
+
+        [JsonProperty("resultProperty")]
+        public StringExpression ResultProperty { get; set; }
+
+        [JsonProperty("cacheTemplate")]
+        public BoolExpression CacheTemplate { get; set; } = true;
+
+        [JsonProperty("cacheExpiration")]
+        public IntExpression CacheExpiration { get; set; } = 900;
+
+        /// <summary>
+        /// Called when the dialog is started and pushed onto the dialog stack.
+        /// </summary>
+        /// <param name="dc">The <see cref="DialogContext"/> for the current turn of conversation.</param>
+        /// <param name="options">Optional, initial information to pass to the dialog.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (options is CancellationToken)
+            {
+                throw new ArgumentException($"{nameof(options)} cannot be a cancellation token");
+            }
+
+            if (this.Disabled != null && this.Disabled.GetValue(dc.State) == true)
+            {
+                return await dc.EndDialogAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+
+            var url = this.Location?.GetValue(dc.State);
+            if (String.IsNullOrEmpty(url))
+            {
+                throw new Exception($"{this.Id}: a template url wasn't provided.");
+            }
+
+            JObject template = null;
+            var cache = this.CacheTemplate?.GetValue(dc.State);
+            if (cache == true)
```

Comment:
VS won't let me do that because it can be null

---
In reply to: [610194282](https://github.com/microsoft/botframework-components/pull/862#discussion_r610194282) [](ancestors = 610194282)